packages/xo-server-sdn-controller/src/index.js

```diff
@@ -446,6 +576,44 @@ class SDNController extends EventEmitter {
         })
         poolNetwork.starCenter = pif.host
         this._starCenters.set(host.$id, host.$ref)
+
+        const crossPoolNetwork = find(
+          this._crossPoolNetworks,
+          crossPoolNetwork =>
+            crossPoolNetwork.networks.includes(pif.$network.$ref)
+        )
+        if (crossPoolNetwork !== undefined) {
+          if (crossPoolNetwork.poolCenter !== undefined) {
+            const centerPoolNetwork = this._getPoolNetwork(
+              crossPoolNetwork.poolCenter,
+              crossPoolNetwork
+            )
+            await this._resetCrossPoolConnections(
+              poolNetwork,
+              crossPoolNetwork.uuid
+            )
+            await this._clearCenterPoolFromNetwork(
+              centerPoolNetwork,
+              poolNetwork,
+              crossPoolNetwork.uuid
+            )
+            await this._connectNetworks(
+              poolNetwork,
+              centerPoolNetwork,
+              crossPoolNetwork.uuid
+            )
+          } else {
+            crossPoolNetwork.poolCenter = host.$pool.$ref
+            log.debug(
+              'First available pool becomes pool center of cross-pool network',
+              {
+                network: pif.$network.$ref,
```

Comment:
`pif.$network.$ref` → `pif.network`.

`$` props are magical because they resolve the refs.